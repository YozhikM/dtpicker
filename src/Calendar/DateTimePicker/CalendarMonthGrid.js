/* @flow */

import React from 'react';
import {
  startOfMonth,
  endOfMonth,
  eachDay,
  getDay,
  isSameDay,
  isWithinRange,
  getDate,
} from 'date-fns';
import s from './Calendar.scss';
import CalendarDay from './CalendarDay';
import format from './format';
import { convertToUTCNewDate, getDateFromValue, type Highlight } from '../helpers';

type Value = {
  year: number,
  month: number,
};

type Week = number[];

export type Props = {|
  value: Value,
  highlight?: Highlight[] | Highlight,
  maxDate?: Date,
  minDate?: Date,
  onSetDate?: (day: number) => void,
  onClickMonth?: (month: Value) => void,
|};

type State = {
  weeks: Week[],
};

export default class CalendarMonthGrid extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      weeks: this.createMonth(props.value) || [],
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.value !== this.props.value) {
      this.setState({ weeks: this.createMonth(nextProps.value) });
    }
  }

  createMonth(date: Value): Week[] {
    const MONTH = new Date(date.year, date.month);
    const startDateOfMonth = startOfMonth(MONTH);
    const endDateOfMonth = endOfMonth(MONTH);
    const currentMonth: number[] = eachDay(startDateOfMonth, endDateOfMonth).map(day =>
      getDate(convertToUTCNewDate(day))
    );

    const firstOfMonthDay: number = getDay(startDateOfMonth);
    let beginOfMonth;
    if (firstOfMonthDay === 0) {
      beginOfMonth = 1;
    } else {
      beginOfMonth = 8 - firstOfMonthDay;
    }
    const firstWeek = currentMonth.slice(0, beginOfMonth);
    const secondWeek = currentMonth.slice(beginOfMonth, 7 + firstWeek.length);
    const thirdWeek = currentMonth.slice(beginOfMonth + 7, beginOfMonth + 14);
    const fourthWeek = currentMonth.slice(beginOfMonth + 14, beginOfMonth + 21);
    const fiveWeek = currentMonth.slice(beginOfMonth + 21, beginOfMonth + 28);
    const tailWeek = currentMonth.slice(beginOfMonth + 28);
    const weeks: Week[] = [firstWeek, secondWeek, thirdWeek, fourthWeek];
    // check for february starting on monday, out of 28 days
    if (fiveWeek.length > 0) {
      weeks.push(fiveWeek);
    }
    if (tailWeek.length > 0) {
      weeks.push(tailWeek);
    }
    return weeks;
  }

  onSetDate = (day: number) => {
    const { maxDate, minDate, onSetDate, value } = this.props;
    const date = new Date(value.year, value.month, day);

    if ((maxDate && date > maxDate) || (minDate && date < minDate)) return;

    if (onSetDate) {
      onSetDate(day);
    }
  };

  onClickMonth = () => {
    const { onClickMonth, value } = this.props;
    if (onClickMonth) {
      onClickMonth(value);
    }
  };

  getWeekDays(): string[] {
    const { value } = this.props;
    const { weeks } = this.state;

    return weeks[1].map(weekDay => {
      const date = new Date(value.year, value.month, weekDay);
      return format(date, 'dd');
    });
  }

  setDayStyle = (highlight: Highlight[] | Highlight, day: number) => {
    const { maxDate, minDate, value } = this.props;
    const date = new Date(value.year, value.month, day);

    const highlightStyle = { backgroundColor: '#34495e', color: '#fff' };
    const rangeStyle = { backgroundColor: '#8196ab', color: '#fff' };
    const disabledDateStyle = { color: 'lightgrey' };

    if (Array.isArray(highlight)) {
      const [first, second] = highlight;
      const firstDate = getDateFromValue(first);
      const secondDate = getDateFromValue(second);

      if ((maxDate && date > maxDate) || (minDate && date < minDate)) {
        return disabledDateStyle;
      }

      for (let i = 0; i < highlight.length; i++) {
        if (isSameDay(getDateFromValue(highlight[i]), date)) {
          return highlightStyle;
        }
      }
      if (firstDate < secondDate) {
        if (isWithinRange(date, firstDate, secondDate)) {
          return rangeStyle;
        }
      } else if (isWithinRange(date, secondDate, firstDate)) {
        return rangeStyle;
      }
    } else if (isSameDay(getDateFromValue(highlight), date)) {
      return highlightStyle;
    }

    return {};
  };

  render() {
    const { weeks } = this.state;
    const { value, highlight } = this.props;
    const weekDays = this.getWeekDays();
    const month = new Date(value.year, value.month);

    return (
      <div className={s.root}>
        <span className={s.span} onClick={this.onClickMonth}>
          {format(month, 'MMMM YYYY')}
        </span>
        <table>
          <thead>
            <tr>
              {weekDays.map(weekDay => {
                return <th key={weekDay}>{weekDay.toUpperCase()}</th>;
              })}
            </tr>
          </thead>

          <tbody>
            {weeks.map(week => (
              <tr key={week[0]}>
                {week.map(day => {
                  return (
                    <CalendarDay
                      key={day}
                      day={day}
                      style={highlight ? this.setDayStyle(highlight, day) : {}}
                      onClick={this.onSetDate}
                    />
                  );
                })}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
