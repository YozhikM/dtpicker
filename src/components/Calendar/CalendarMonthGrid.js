/* @flow */

import React from 'react';
import s from './Calendar.scss';
import Day from './Day/Day';
import { startOfMonth, endOfMonth, eachDay, getDay, format } from 'date-fns';

export type Props = {
  date: Date,
  activeDates: Date,
  onClickDay?: (date: Date) => void,
  onClickMonth?: (date: Date) => void
};

type State = {
  weeks: Array<any>
};

class CalendarMonthGrid extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      weeks: []
    };
  }

  componentWillMount() {
    const { date } = this.props;
    this.setState({ weeks: this.createMonth(date) });
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.date !== this.props.date) {
      this.setState({ weeks: this.createMonth(nextProps.date) });
    }
  }

  createMonth = (date: Date) => {
    let startDateOfMonth = startOfMonth(date);
    let endDateOfMonth = endOfMonth(date);
    let currentMonth = eachDay(startDateOfMonth, endDateOfMonth);
    let firstOfMonthDay = getDay(startDateOfMonth);
    let beginOfMonth;
    firstOfMonthDay === 0 ? (beginOfMonth = 1) : (beginOfMonth = 8 - firstOfMonthDay);
    let firstWeek = currentMonth.slice(0, beginOfMonth);
    let secondWeek = currentMonth.slice(beginOfMonth, 7 + firstWeek.length);
    let thirdWeek = currentMonth.slice(beginOfMonth + 7, beginOfMonth + 14);
    let fourthWeek = currentMonth.slice(beginOfMonth + 14, beginOfMonth + 21);
    let fiveWeek = currentMonth.slice(beginOfMonth + 21, beginOfMonth + 28);
    let tailWeek = currentMonth.slice(beginOfMonth + 28);
    const weeks = [firstWeek, secondWeek, thirdWeek, fourthWeek, fiveWeek];
    if (tailWeek.length > 0) {
      weeks.push(tailWeek);
    }
    return weeks;
  };

  isSameDate(lDate: Date, rDate: Date): boolean {
    return (
      lDate.getDate() === rDate.getDate() &&
      lDate.getMonth() === rDate.getMonth() &&
      lDate.getFullYear() === rDate.getFullYear()
    );
  }

  onClickDay = (date: Date) => {
    const { onClickDay } = this.props;
    if (onClickDay) {
      onClickDay(date);
    }
  };

  onClickMonth = () => {
    const { onClickMonth } = this.props;
    if (onClickMonth) {
      onClickMonth(this.props.date);
    }
  };

  checkInvalidDate = () => {
    const { date } = this.props;
    const validFormat = format(date, 'MMMM YYYY');
    if (validFormat === 'Invalid Date') {
      return 'Данные неверны';
    }
    else return validFormat;
  };

  render() {
    const { weeks } = this.state;
    const { activeDates, date } = this.props;

    return (
      <div className={s.root}>
        <span onClick={this.onClickMonth}>{format(date, 'MMMM YYYY')}</span>
        <table>
          <thead>
            <tr>
              <th>ПН</th>
              <th>ВТ</th>
              <th>СР</th>
              <th>ЧТ</th>
              <th>ПТ</th>
              <th>СБ</th>
              <th>ВС</th>
            </tr>
          </thead>
          <tbody>
            {weeks.map((week, i) => (
              <tr key={i}>
                {week.map((date, dayOfWeek) => {
                  return (
                    <Day
                      key={dayOfWeek}
                      date={date}
                      onClick={this.onClickDay}
                      isActive={this.isSameDate(activeDates, date)}
                    />
                  );
                })}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export default CalendarMonthGrid;
