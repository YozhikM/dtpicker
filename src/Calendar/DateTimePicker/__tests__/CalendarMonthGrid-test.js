/* @flow */

import React from 'react';
import CalendarMonthGrid from '../CalendarMonthGrid';

describe('CalendarMonthGrid', () => {
  describe('render()', () => {
    it('basic', () => {
      expect(shallow(<CalendarMonthGrid value={new Date(2009, 0, 9)} />)).toMatchSnapshot();
    });
  });

  it('componentWillMount()', () => {
    const value = new Date(2019, 0, 9);
    const wrapper = shallow(<CalendarMonthGrid value={value} />);
    wrapper.instance().createMonth(value);

    expect(wrapper.state('weeks')).toMatchSnapshot();
  });

  it('componentWillReceiveProps()', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date()} />);
    const nextProps = {
      value: new Date(2017, 10, 12),
      highlight: new Date(),
    };
    wrapper.instance().componentWillReceiveProps(nextProps);

    expect(wrapper.state('weeks')).toMatchSnapshot();
  });

  it('createMonth()', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date()} />);
    const weeks = wrapper.instance().createMonth(new Date(2009, 9, 8));

    expect(weeks).toMatchSnapshot();
  });

  it('getWeekDays()', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date()} />);
    const weekDays = wrapper.instance().getWeekDays();

    expect(weekDays).toMatchSnapshot();
  });

  describe('setDayStyle()', () => {
    it('should be highlightStyle if hightlight is array', () => {
      const wrapper = shallow(<CalendarMonthGrid value={new Date(1994, 6)} />);
      const highlight = [
        new Date('1994-07-01T18:00:00.000Z'),
        new Date('1994-07-05T18:00:00.000Z'),
      ];
      const dayStyle = wrapper.instance().setDayStyle(highlight, highlight[0]);

      expect(dayStyle).toMatchSnapshot();
    });

    it('should be highlightStyle if highlight is date', () => {
      const wrapper = shallow(<CalendarMonthGrid value={new Date(1994, 6)} />);
      const highlight = new Date('1994-07-05T18:00:00.000Z');
      const dayStyle = wrapper
        .instance()
        .setDayStyle(highlight, new Date('1994-07-05T17:59:59.000Z'));

      expect(dayStyle).toMatchSnapshot();
    });

    it('should be rangeStyle', () => {
      const wrapper = shallow(<CalendarMonthGrid value={new Date(1994, 6)} />);
      const highlight = [
        new Date('1994-07-05T18:00:00.000Z'),
        new Date('1994-07-07T18:00:00.000Z'),
      ];
      const dayStyle = wrapper
        .instance()
        .setDayStyle(highlight, new Date('1994-07-06T17:59:59.000Z'));

      expect(dayStyle).toMatchSnapshot();
    });
  });

  it('formatted current value in MMMM YYYY', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date(2018, 2, 11)} />);
    const formattedCurrValue = wrapper.find('span').text();

    expect(formattedCurrValue).toBe('March 2018');
  });

  describe('user interactions', () => {
    it('onSetDate()', () => {
      const onSetDate = jest.fn();
      const wrapper = shallow(
        <CalendarMonthGrid value={new Date(2010, 0)} onSetDate={onSetDate} />
      );
      const newDate = new Date(2017, 1, 1);
      wrapper
        .find('CalendarDay')
        .at(0)
        .props()
        .onClick(newDate);

      expect(onSetDate).toBeCalledWith(newDate);
    });

    it('onClickMonth()', () => {
      const onClickMonth = jest.fn();
      const value = new Date(2009, 1);
      const wrapper = shallow(<CalendarMonthGrid value={value} onClickMonth={onClickMonth} />);
      wrapper.find('span').simulate('click');

      expect(onClickMonth).toBeCalledWith(value);
    });
  });

  describe('test cases', () => {
    it('check for february starting on monday, out of 28 days', () => {
      const wrapper = shallow(<CalendarMonthGrid value={new Date(2010, 1, 2)} />);

      expect(wrapper.state('weeks')).toHaveLength(4);
    });
  });
});
