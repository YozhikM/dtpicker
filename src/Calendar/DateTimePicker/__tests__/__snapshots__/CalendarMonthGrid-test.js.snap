// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarMonthGrid componentWillMount() 1`] = `
Array [
  Array [
    2018-12-31T18:00:00.000Z,
    2019-01-01T18:00:00.000Z,
    2019-01-02T18:00:00.000Z,
    2019-01-03T18:00:00.000Z,
    2019-01-04T18:00:00.000Z,
    2019-01-05T18:00:00.000Z,
  ],
  Array [
    2019-01-06T18:00:00.000Z,
    2019-01-07T18:00:00.000Z,
    2019-01-08T18:00:00.000Z,
    2019-01-09T18:00:00.000Z,
    2019-01-10T18:00:00.000Z,
    2019-01-11T18:00:00.000Z,
    2019-01-12T18:00:00.000Z,
  ],
  Array [
    2019-01-13T18:00:00.000Z,
    2019-01-14T18:00:00.000Z,
    2019-01-15T18:00:00.000Z,
    2019-01-16T18:00:00.000Z,
    2019-01-17T18:00:00.000Z,
    2019-01-18T18:00:00.000Z,
    2019-01-19T18:00:00.000Z,
  ],
  Array [
    2019-01-20T18:00:00.000Z,
    2019-01-21T18:00:00.000Z,
    2019-01-22T18:00:00.000Z,
    2019-01-23T18:00:00.000Z,
    2019-01-24T18:00:00.000Z,
    2019-01-25T18:00:00.000Z,
    2019-01-26T18:00:00.000Z,
  ],
  Array [
    2019-01-27T18:00:00.000Z,
    2019-01-28T18:00:00.000Z,
    2019-01-29T18:00:00.000Z,
    2019-01-30T18:00:00.000Z,
  ],
]
`;

exports[`CalendarMonthGrid componentWillReceiveProps() 1`] = `
Array [
  Array [
    2017-10-31T18:00:00.000Z,
    2017-11-01T18:00:00.000Z,
    2017-11-02T18:00:00.000Z,
    2017-11-03T18:00:00.000Z,
    2017-11-04T18:00:00.000Z,
  ],
  Array [
    2017-11-05T18:00:00.000Z,
    2017-11-06T18:00:00.000Z,
    2017-11-07T18:00:00.000Z,
    2017-11-08T18:00:00.000Z,
    2017-11-09T18:00:00.000Z,
    2017-11-10T18:00:00.000Z,
    2017-11-11T18:00:00.000Z,
  ],
  Array [
    2017-11-12T18:00:00.000Z,
    2017-11-13T18:00:00.000Z,
    2017-11-14T18:00:00.000Z,
    2017-11-15T18:00:00.000Z,
    2017-11-16T18:00:00.000Z,
    2017-11-17T18:00:00.000Z,
    2017-11-18T18:00:00.000Z,
  ],
  Array [
    2017-11-19T18:00:00.000Z,
    2017-11-20T18:00:00.000Z,
    2017-11-21T18:00:00.000Z,
    2017-11-22T18:00:00.000Z,
    2017-11-23T18:00:00.000Z,
    2017-11-24T18:00:00.000Z,
    2017-11-25T18:00:00.000Z,
  ],
  Array [
    2017-11-26T18:00:00.000Z,
    2017-11-27T18:00:00.000Z,
    2017-11-28T18:00:00.000Z,
    2017-11-29T18:00:00.000Z,
  ],
]
`;

exports[`CalendarMonthGrid createMonth() 1`] = `
Array [
  Array [
    2009-09-30T18:00:00.000Z,
    2009-10-01T18:00:00.000Z,
    2009-10-02T18:00:00.000Z,
    2009-10-03T18:00:00.000Z,
  ],
  Array [
    2009-10-04T18:00:00.000Z,
    2009-10-05T18:00:00.000Z,
    2009-10-06T18:00:00.000Z,
    2009-10-07T18:00:00.000Z,
    2009-10-08T18:00:00.000Z,
    2009-10-09T18:00:00.000Z,
    2009-10-10T18:00:00.000Z,
  ],
  Array [
    2009-10-11T18:00:00.000Z,
    2009-10-12T18:00:00.000Z,
    2009-10-13T18:00:00.000Z,
    2009-10-14T18:00:00.000Z,
    2009-10-15T18:00:00.000Z,
    2009-10-16T18:00:00.000Z,
    2009-10-17T18:00:00.000Z,
  ],
  Array [
    2009-10-18T18:00:00.000Z,
    2009-10-19T18:00:00.000Z,
    2009-10-20T18:00:00.000Z,
    2009-10-21T18:00:00.000Z,
    2009-10-22T18:00:00.000Z,
    2009-10-23T18:00:00.000Z,
    2009-10-24T18:00:00.000Z,
  ],
  Array [
    2009-10-25T18:00:00.000Z,
    2009-10-26T18:00:00.000Z,
    2009-10-27T18:00:00.000Z,
    2009-10-28T18:00:00.000Z,
    2009-10-29T18:00:00.000Z,
    2009-10-30T18:00:00.000Z,
  ],
]
`;

exports[`CalendarMonthGrid getWeekDays() 1`] = `
Array [
  "Mo",
  "Tu",
  "We",
  "Th",
  "Fr",
  "Sa",
  "Su",
]
`;

exports[`CalendarMonthGrid render() basic 1`] = `
<div>
  <span
    onClick={[Function]}
  >
    January 2009
  </span>
  <table>
    <thead>
      <tr>
        <th
          key="Mo"
        >
          MO
        </th>
        <th
          key="Tu"
        >
          TU
        </th>
        <th
          key="We"
        >
          WE
        </th>
        <th
          key="Th"
        >
          TH
        </th>
        <th
          key="Fr"
        >
          FR
        </th>
        <th
          key="Sa"
        >
          SA
        </th>
        <th
          key="Su"
        >
          SU
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="2008-12-31T18:00:00.000Z"
      >
        <CalendarDay
          date={2008-12-31T18:00:00.000Z}
          key="2008-12-31T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-01T18:00:00.000Z}
          key="2009-01-01T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-02T18:00:00.000Z}
          key="2009-01-02T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-03T18:00:00.000Z}
          key="2009-01-03T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
      </tr>
      <tr
        key="2009-01-04T18:00:00.000Z"
      >
        <CalendarDay
          date={2009-01-04T18:00:00.000Z}
          key="2009-01-04T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-05T18:00:00.000Z}
          key="2009-01-05T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-06T18:00:00.000Z}
          key="2009-01-06T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-07T18:00:00.000Z}
          key="2009-01-07T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-08T18:00:00.000Z}
          key="2009-01-08T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-09T18:00:00.000Z}
          key="2009-01-09T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-10T18:00:00.000Z}
          key="2009-01-10T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
      </tr>
      <tr
        key="2009-01-11T18:00:00.000Z"
      >
        <CalendarDay
          date={2009-01-11T18:00:00.000Z}
          key="2009-01-11T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-12T18:00:00.000Z}
          key="2009-01-12T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-13T18:00:00.000Z}
          key="2009-01-13T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-14T18:00:00.000Z}
          key="2009-01-14T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-15T18:00:00.000Z}
          key="2009-01-15T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-16T18:00:00.000Z}
          key="2009-01-16T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-17T18:00:00.000Z}
          key="2009-01-17T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
      </tr>
      <tr
        key="2009-01-18T18:00:00.000Z"
      >
        <CalendarDay
          date={2009-01-18T18:00:00.000Z}
          key="2009-01-18T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-19T18:00:00.000Z}
          key="2009-01-19T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-20T18:00:00.000Z}
          key="2009-01-20T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-21T18:00:00.000Z}
          key="2009-01-21T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-22T18:00:00.000Z}
          key="2009-01-22T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-23T18:00:00.000Z}
          key="2009-01-23T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-24T18:00:00.000Z}
          key="2009-01-24T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
      </tr>
      <tr
        key="2009-01-25T18:00:00.000Z"
      >
        <CalendarDay
          date={2009-01-25T18:00:00.000Z}
          key="2009-01-25T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-26T18:00:00.000Z}
          key="2009-01-26T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-27T18:00:00.000Z}
          key="2009-01-27T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-28T18:00:00.000Z}
          key="2009-01-28T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-29T18:00:00.000Z}
          key="2009-01-29T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
        <CalendarDay
          date={2009-01-30T18:00:00.000Z}
          key="2009-01-30T18:00:00.000Z"
          onClick={[Function]}
          style={Object {}}
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CalendarMonthGrid setDayStyle() should be highlightStyle if highlight is date 1`] = `
Object {
  "backgroundColor": "#34495e",
  "color": "#fff",
}
`;

exports[`CalendarMonthGrid setDayStyle() should be highlightStyle if hightlight is array 1`] = `
Object {
  "backgroundColor": "#34495e",
  "color": "#fff",
}
`;

exports[`CalendarMonthGrid setDayStyle() should be rangeStyle 1`] = `
Object {
  "backgroundColor": "#8196ab",
  "color": "#fff",
}
`;
